File: typical90/030/main.cpp
#include <iostream>
#include <vector>

using namespace std;

// 素数を求める関数
vector<int> getPrimes(int n) {
    vector<bool> is_prime(n + 1, true);
    is_prime[0] = false;
    is_prime[1] = false;
    for (int i = 2; i * i <= n; i++) {
        if (!is_prime[i]) continue;
        for (int j = i * 2; j <= n; j += i) {
            is_prime[j] = false;
        }
    }

    vector<int> primes;
    for (int i = 2; i <= n; i++) {
        if (is_prime[i]) primes.push_back(i);
    }
    return primes;
}

int main() {
    int N, K;
    cin >> N >> K;

    vector<int> primes = getPrimes(N);
    vector<int> S(N + 1, 0);

    for (int p : primes) {
        for (int bp = p; bp <= N; bp += p) {
            S[bp]++;
        }
    }

    int ans = 0;
    for (int s : S) {
        if (s >= K) {
            ans++;
        }
    }

    cout << ans << endl;
    return 0;
}

File: typical90/030/main.py
N, K = map(int, input().split())
# A_list = list(map(int, input().split()))

def getprimes(n):
    is_prime = [True] * (n + 1)
    is_prime[0] = False
    is_prime[1] = False
    for i in range(2, int(n**0.5) + 1):
        if not is_prime[i]:
            continue
        for j in range(i * 2, n + 1, i):
            is_prime[j] = False
    return [i for i in range(n + 1) if is_prime[i]]

primes = getprimes(N)

S = [0]*(N+1)

for p in primes:
    for bp in range(p, N+1, p):
        S[bp] += 1

ans = 0
for s in S:
    if s >= K:
        ans += 1

print(ans)

